frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -282,14 +282,9 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
    */
   const ConditionalEditButton = () => (
     <>
-      {disabled &&
-        (keycloak.hasClaim(Claims.ADMIN_PROPERTIES) ||
-          (keycloak.hasClaim(Claims.PROPERTY_EDIT) &&
-            ((parcelDetail?.agencyId && keycloak.agencyIds.includes(parcelDetail.agencyId)) ||
-              (buildingDetail?.agencyId &&
-                keycloak.agencyIds.includes(buildingDetail.agencyId))))) && (
-          <EditButton onClick={() => setDisabled(false)} />
-        )}
+      {disabled && keycloak.canUserEditProperty(buildingDetail ?? parcelDetail) && (
```

Comment:
using the fact that buildingDetail and parcelDetail both inherit from IProperty here, and IProperty has both the agencyId and projectNumber fields. This means that I can just pass in either of the above as an IProperty and do all comparisons on the one object.

Reply:
also moved the comparison to a common keycloak function since it will be used in a lot of places.